uint8 number
---
bool success
